sdk/storage/azure-storage-file-share/tests/test_share.py

```diff
@@ -133,44 +144,52 @@ def test_create_snapshot_with_metadata(self, storage_account_name, storage_accou
         )
         snapshot_props = snapshot_client.get_share_properties()
         # Assert
-        self.assertTrue(created)
-        self.assertIsNotNone(snapshot['snapshot'])
-        self.assertIsNotNone(snapshot['etag'])
-        self.assertIsNotNone(snapshot['last_modified'])
-        self.assertEqual(share_props.metadata, metadata)
-        self.assertEqual(snapshot_props.metadata, metadata2)
+        assert created
+        assert snapshot['snapshot'] is not None
+        assert snapshot['etag'] is not None
+        assert snapshot['last_modified'] is not None
+        assert share_props.metadata == metadata
+        assert snapshot_props.metadata == metadata2
         self._delete_shares(share.share_name)
 
     @FileSharePreparer()
-    def test_delete_share_with_snapshots(self, storage_account_name, storage_account_key):
+    @recorded_by_proxy
+    def test_delete_share_with_snapshots(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
         self._setup(storage_account_name, storage_account_key)
         share = self._get_share_reference()
         share.create_share()
         snapshot = share.create_snapshot()
 
         # Act
-        with self.assertRaises(HttpResponseError):
+        with pytest.raises(HttpResponseError):
             share.delete_share()
 
         deleted = share.delete_share(delete_snapshots=True)
-        self.assertIsNone(deleted)
+        assert deleted is None
         self._delete_shares()
 
     @pytest.mark.playback_test_only
```

Comment:
I assume we didn't have a file-share soft-delete account int he ARM template?

Reply:
Correct. I tried (albeit, for only ~5 minutes before giving up) to do the pattern-matching necessary but ended up deciding it wasn't worth the effort for 1 test (and it wasn't already nicely hooked up and just not exposed like in `datalake` 😢)